tests/unit/expire.tcl

```diff
@@ -833,6 +833,24 @@ start_server {tags {"expire"}} {
         assert_equal [r debug set-active-expire 1] {OK}
     } {} {needs:debug}
 
+    test {import-source can be closed when import-mode is off} {
+        r flushall
+        set id [r client id]
+
+        catch {r client import-source on} err
+        assert_match {ERR*} $err
+
+        r config set import-mode yes
+        assert_equal [r client import-source on] {OK}
+
+        assert_match {*flags=I*} [r client list id $id]
+
+        r config set import-mode no
+        assert_equal [r client import-source off] {OK}
+
+        assert_match {*flags=N*} [r client list id $id]
+    }
```

Comment:
we can save some lines since it is a very simple test 
```suggestion
    test {import-source can be closed when import-mode is off} {
        r config set import-mode no
        assert_error "ERR Server is not in import mode" {r client import-source on}

        r config set import-mode yes
        assert_equal [r client import-source on] {OK}
        assert_match {*flags=I*} [r client list id [r client id]]

        r config set import-mode no
        assert_equal [r client import-source off] {OK}
        assert_match {*flags=N*} [r client list id [r client id]]
    }
```

Reply:
Good point, we can skip `flushall` and `client id`.

I suggested earlier to match the error only on `ERR*` instead of matching on the full error message. IMO we don't need to match the full message because we can allow it to change.

Reply:
OK, good point that we want to check that it's not a syntax error. It's unfortunate that most of the errors are just ERR. We can't change it now, but it would have been better if each error had its own code, like SYNTAX for syntax error, etc.